src/lcm/typing.py

```diff
@@ -33,24 +33,62 @@ def __call__(  # noqa: D102
     ) -> Scalar: ...
 
 
-class DiscreteProblemValueSolverFunction(Protocol):
-    """The function that solves for the value of the discrete problem.
+class MaxQOverCFunction(Protocol):
```

Comment:
Do we need the `Function` part under the new convention?

Reply:
It is consistent with the other function types, and I like that it is directly clear that this type references a function.

Reply:
> It is consistent with the other function types, and I like that it is directly clear that this type references a function.

Fair enough. I thought it was clear enough, but all good if you prefer it this way. 

*(and apologies for not writing that my suggestion should have applied to all "Functions" in this module)*